### PR TITLE
[All Hosts] (debugging) rewrite of article about debugging in Visual Studio

### PR DESCRIPTION
**Pull request by Rick-Kirkham**
It has not been possible to debug client-side code in Visual Studio for years. This article was badly out-of-date. It has been rewritten to show how to use the developer tools to debug. I've also moved the article to the debugging section of the table of contents where all the other debugging articles are. And I added a note to deal with the fact that the manifest XSDs that Visual Studio installs are out-of-date.

### CODE REVIEW

**Review by igor-ribeiiro — APPROVED** [review]
Looks great! Should we also add if Project and Visio are supported?
We have some other documentation on them and I feel they shouldn't be left out

**Comment by Rick-Kirkham** [comment]
> Looks great! Should we also add if Project and Visio are supported? We have some other documentation on them and I feel they shouldn't be left out

@igor-ribeiiro There are no Visual Studio project templates for Project or Visio. (And as far as I know, there's never been any customer demand for them.)

File: docs/develop/debug-office-add-ins-in-visual-studio.md [hunk 1]
```diff
@@ -68,89 +60,133 @@ The following table describes the properties of the web application project that
...
+This section describes how to start and debug an Outlook add-in.
+
+### Start the Outlook add-in project
+
+Start the project by choosing **Debug** > **Start Debugging** from the menu bar or press the F5 button. Visual Studio will automatically build the solution and launch the Outlook page of your Microsoft 365 tenancy.
 
 When Visual Studio builds the project it performs the following tasks.
 
-1. Creates a copy of the XML manifest file and adds it to  `_ProjectName_\bin\Debug\OfficeAppManifests` directory. The Office application that hosts your add-in consumes this copy when you start Visual Studio and debug the add-in.
+1. Prompts you for login credentials. If you're asked to sign in repeatedly or if you receive an error that you are unauthorized, then Basic Auth may be disabled for accounts on your Microsoft 365 tenant. In this case, try using a Microsoft account instead. You can also try setting the property **Use multi-factor auth** to **True** in the Outlook Web Add-in project properties dialog.
```
Comment (samantharamon) [inline docs/develop/debug-office-add-ins-in-visual-studio.md:129]: Consider linking to the 'Add-in project properties' section (https://review.docs.microsoft.com/en-us/office/dev/add-ins/develop/debug-office-add-ins-in-visual-studio?branch=pr-en-us-3253#add-in-project-properties).

File: docs/develop/debug-office-add-ins-in-visual-studio.md [hunk 2]
```diff
@@ -68,89 +60,133 @@ The following table describes the properties of the web application project that
...
 
-2. Creates a set of registry entries on your computer that enable the add-in to appear in the Office application.
+1. Creates a copy of the XML manifest file and adds it to  `_ProjectName_\bin\Debug\OfficeAppManifests` directory. Outlook consumes this copy when you start Visual Studio and debug the add-in.
```
Comment (samantharamon) [inline docs/develop/debug-office-add-ins-in-visual-studio.md:131]:
```suggestion
1. Creates a copy of the XML manifest file and adds it to the `_ProjectName_\bin\Debug\OfficeAppManifests` directory. Outlook consumes this copy when you start Visual Studio and debug the add-in.
```

File: docs/develop/debug-office-add-ins-in-visual-studio.md [hunk 3]
```diff
@@ -68,89 +60,133 @@ The following table describes the properties of the web application project that
...
 
-3. Builds the web application project, and then deploys it to the local IIS web server (https://localhost).
+2. Builds the web application project, and then deploys it to the local IIS web server (https://localhost).
```
Comment (samantharamon) [inline docs/develop/debug-office-add-ins-in-visual-studio.md:133]:
```suggestion
2. Builds the web application project, and then deploys it to the local IIS web server (`https://localhost`).
```

File: docs/develop/debug-office-add-ins-in-visual-studio.md [hunk 4]
```diff
@@ -68,89 +60,133 @@ The following table describes the properties of the web application project that
...
 
-4. If this is the first add-in project that you have deployed to local IIS web server, you may be prompted to install a Self-Signed Certificate to the current user's Trusted Root Certificate store. This is required for IIS Express to display the content of your add-in correctly.
+3. If this is the first add-in project that you have deployed to local IIS web server, you may be prompted to install a Self-Signed Certificate to the current user's Trusted Root Certificate store. This is required for IIS Express to display the content of your add-in correctly.
```
Comment (samantharamon) [inline docs/develop/debug-office-add-ins-in-visual-studio.md:135]:
```suggestion
3. If this is the first add-in project that you have deployed to the local IIS web server, you may be prompted to install a Self-Signed Certificate to the current user's Trusted Root Certificate store. This is required for IIS Express to display the content of your add-in correctly.
```

File: docs/develop/debug-office-add-ins-in-visual-studio.md [hunk 5]
```diff
@@ -68,89 +60,133 @@ The following table describes the properties of the web application project that
...
+1. Creates a copy of the XML manifest file and adds it to  `_ProjectName_\bin\Debug\OfficeAppManifests` directory. The Office application that hosts your add-in consumes this copy when you start Visual Studio and debug the add-in.
 
-5. In **Solution Explorer**, choose the add-in project (*not* the web application project).
+2. Creates a set of registry entries on your Windows computer that enable the add-in to appear in the Office application.
 
-6. From the menu bar, choose **View** > **Properties Window**.
+3. Builds the web application project, and then deploys it to the local IIS web server (https://localhost).
 
-7. In the **Properties** window, choose the **Start Document** list, and then select the document that you added to the project. The project is now configured to start the add-in in that document.
+4. If this is the first add-in project that you have deployed to local IIS web server, you may be prompted to install a Self-Signed Certificate to the current user's Trusted Root Certificate store. This is required for IIS Express to display the content of your add-in correctly.
```
Comment (samantharamon) [inline docs/develop/debug-office-add-ins-in-visual-studio.md:79]:
```suggestion
4. If this is the first add-in project that you have deployed to the local IIS web server, you may be prompted to install a Self-Signed Certificate to the current user's Trusted Root Certificate store. This is required for IIS Express to display the content of your add-in correctly.
```

File: docs/develop/debug-office-add-ins-in-visual-studio.md [hunk 6]
```diff
@@ -68,89 +60,133 @@ The following table describes the properties of the web application project that
...
 
 > [!NOTE]
-> The latest version of Office may use a newer web control to display the add-in contents when running on Windows 10. If this is the case, Visual Studio may prompt you to add a local network loopback exemption. This is required for the web control, in the Office client application, to be able to access the website deployed to the local IIS web server. You can also change this setting anytime in Visual Studio under **Tools** > **Options** > **Office Tools (Web)** > **Web Add-In Debugging**.
+> If Office uses the Edge Legacy webview control (EdgeHTML) to run add-ins on your Windows computer, Visual Studio may prompt you to add a local network loopback exemption. This is required for the webview control to be able to access the website deployed to the local IIS web server. You can also change this setting anytime in Visual Studio under **Tools** > **Options** > **Office Tools (Web)** > **Web Add-In Debugging**. To find out what browser control is used on your Windows computer, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).
```
Comment (samantharamon) [inline docs/develop/debug-office-add-ins-in-visual-studio.md:138]:
```suggestion
> If Office uses the Edge Legacy WebView control (EdgeHTML) to run add-ins on your Windows computer, Visual Studio may prompt you to add a local network loopback exemption. This is required for the WebView control to be able to access the website deployed to the local IIS web server. You can also change this setting anytime in Visual Studio under **Tools** > **Options** > **Office Tools (Web)** > **Web Add-In Debugging**. To find out what browser control is used on your Windows computer, see [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).
```

Comment (samantharamon) [reply docs/develop/debug-office-add-ins-in-visual-studio.md:138]: As this note block is referenced more than once in the article, consider creating an include file for it.

File: docs/develop/debug-office-add-ins-in-visual-studio.md [hunk 7]
```diff
@@ -68,89 +60,133 @@ The following table describes the properties of the web application project that
...
+1. Modifies the [SourceLocation](../reference/manifest/sourcelocation.md) element of the XML manifest file (that was copied to the `_ProjectName_\bin\Debug\OfficeAppManifests` directory) by replacing the `~remoteAppUrl` token with the fully qualified address of the start page (for example, `https://localhost:44302/Home.html`).
 
 2. Starts the web application project in IIS Express.
 
-3. Opens the Office application.
+3. Validates the manifest. To review the validation rules of the XML manifest file in your project, see [Office Add-ins XML manifest](../develop/add-in-manifests.md). 
+
+   > [!IMPORTANT]
+   > The Office manifest XSD files that Visual Studio installs are out-of-date. If you get validation errors for the manifest, your first troubleshooting step should be to replace one or more of these files with the latest versions. For detailed instructions, see [Manifest schema validation errors in Visual Studio projects](../testing/troubleshoot-development-errors.md#manifest-schema-validation-errors-in-visual-studio-projects).
```
Comment (samantharamon) [inline docs/develop/debug-office-add-ins-in-visual-studio.md:149]: As this note block is referenced more than once in the article, consider creating an include file for it.

File: docs/develop/debug-office-add-ins-in-visual-studio.md [hunk 8]
```diff
@@ -68,89 +60,133 @@ The following table describes the properties of the web application project that
...
 
-Visual Studio doesn't show validation errors in the **OUTPUT** window when you build the project. Visual Studio reports errors and warnings in the **ERRORLIST** window as they occur. Visual Studio also reports validation errors by showing wavy underlines (known as squiggles) of different colors in the code and text editor. These marks notify you of problems that Visual Studio detected in your code. For more information about how to enable or disable validation, see [Options, Text Editor, JavaScript, IntelliSense](/visualstudio/ide/reference/options-text-editor-javascript-intellisense?view=vs-2019&preserve-view=true).
+4. Opens the Outlook page of your Microsoft 365 tenancy in the Microsoft Edge.
```
Comment (samantharamon) [inline docs/develop/debug-office-add-ins-in-visual-studio.md:151]:
```suggestion
4. Opens the Outlook page of your Microsoft 365 tenancy in Microsoft Edge.
```

File: docs/develop/debug-office-add-ins-in-visual-studio.md [hunk 9]
```diff
@@ -68,89 +60,133 @@ The following table describes the properties of the web application project that
...
+### Debug the Outlook add-in
 
-## Debug the code for an Excel, PowerPoint, or Word add-in
+1. In the Outlook page, select an email message or appointment item to open it in its own window. 
 
-If your add-in isn't visible within the document that's displayed in the Office application (Excel, PowerPoint, or Word) after you've [started the project](#start-the-project), manually launch the add-in in the Office application. For example, launch your task pane add-in by choosing the **Show Taskpane** button in the ribbon of the **Home** tab. After your add-in is displayed in Excel, PowerPoint, or Word, you can debug your code by doing the following:
+2. Press F12 to open the Edge debugging tool.
 
-1. In Excel, PowerPoint, or Word, choose the **Insert** tab and then choose the down-arrow located to the right of **My Add-ins**.
+3. After the tool is open, launch the add-in. For example, in the toolbar at the top of a message, select the **More apps** button and then select your add-in from the callout that opens.
```
Comment (samantharamon) [inline docs/develop/debug-office-add-ins-in-visual-studio.md:159]:
```suggestion
3. After the tool is open, launch the add-in. For example, in the toolbar at the top of a message, select the **More apps** button, and then select your add-in from the callout that opens.
```

File: docs/develop/debug-office-add-ins-in-visual-studio.md [hunk 10]
```diff
@@ -68,89 +60,133 @@ The following table describes the properties of the web application project that
...
-2. In the list of available add-ins, find the **Developer Add-ins** section and select the your add-in to register it.
+4. Use the instructions in one of the following articles to set breakpoints and step through code. They each have a link to more detailed guidance.
 
-3. In Visual Studio, set breakpoints in your code.
+   - [Debug add-ins using developer tools for Edge Legacy](../testing/debug-add-ins-using-devtools-edge-legacy.md)
+   - [Debug add-ins using developer tools in Microsoft Edge (Chromium-based)](../testing/debug-add-ins-using-devtools-edge-chromium.md)
 
-4. In Excel, PowerPoint, or Word, interact with your add-in.
+   > [!TIP]
+   > To debug code that runs in the `Office.initialize` method or an `Office.onReady` method that runs when the add-in opens, set your breakpoints and then close and reopen the add-in. For more information about these methods, see [Initialize your Office Add-in](../develop/initialize-add-in.md).
```
Comment (samantharamon) [inline docs/develop/debug-office-add-ins-in-visual-studio.md:169]:
```suggestion
   > To debug code that runs in the `Office.initialize` method or an `Office.onReady` method that runs when the add-in opens, set your breakpoints, and then close and reopen the add-in. For more information about these methods, see [Initialize your Office Add-in](../develop/initialize-add-in.md).
```

**Review by samantharamon — APPROVED** [review]
Great stuff! Approved with minor suggestions.

**Pull request merged**